PinpointKit/PinpointKit/Sources/ScreenshotDetector/ScreenshotDetector.swift

```diff
@@ -63,17 +63,36 @@ open class ScreenshotDetector: NSObject {
     }
     
     private func requestPhotosAuthorization() {
-        PHPhotoLibrary.requestAuthorization { authorizationStatus in
-            OperationQueue.main.addOperation {
-                switch authorizationStatus {
-                case .authorized:
-                    // Register for the next photo library change notification since the new screenshot
-                    // won’t be available immediately.
-                    self.photoLibrary.register(self)
-                case .denied, .notDetermined, .restricted:
-                    self.fail(with: .unauthorized(status: authorizationStatus))
-                @unknown default:
-                    break
+        
+        func handleStatus(status: PHAuthorizationStatus) {
+            switch status {
+            case .authorized:
+                // Register for the next photo library change notification since the new screenshot
+                // won’t be available immediately.
+                self.photoLibrary.register(self)
+            case .limited:
```

Comment:
On first install, if the user selects the limited option, but dismisses the picker that lets them select photos to share with the app by swiping it down (not tapping Cancel), PinpointKit doesn’t present on future screenshots taken during this launch. Not sure what can be done here, whether authorization is denied or if this can be worked around, but it feels like a bug in usage.

Related, if you tap cancel on the photo picker (again, on first install, when prompted and you choose to select photos), PinpointKit _does_ present after canceling. This may be considered intended or a bug, I‘m not really sure, since we don’t use the selection from this step anyway.

Reply:
What are your thoughts here @Twigz?